### ISSUE TITLE
winforms 2.0 for linux - mono -> dotnet.core 

### ISSUE THREAD

**Issue by sharkguto** [dotnet/winforms#3750]
it would be great if microsoft copy and paste winforms 2.0 from mono to dotnet core.

**Comment by italocassio** [comment]
Yes!!! that would be incredible!
Go go Microsoft!

**Comment by OliaG** [comment]
Simple copy-pasting unfortunately would not work and it will require enormous resources to bring it to a workable and quality satisfactory state. So unfortunately as of now WinForms will remain Windows only. But we are working on a cross-platform desktop solution .NET MAUI.

**Comment by sharkguto** [comment]
https://devblogs.microsoft.com/dotnet/introducing-net-multi-platform-app-ui/

**Comment by elachlan** [comment]
I do not want yet another mobile UI framework extended to desktop. There are many productivity applications that do not lend themselves to mobile based design. 

Correct me if I am wrong, but if MAUI is inheriting from xamarin.forms then it won't have multiple windows. A pretty big requirement for a lot of applications.

I just want my winforms apps to run on mac so I can natively support it instead of getting customers to go through the pain that is parallels.

**Comment by sharkguto** [comment]
@elachlan its really sad that microsoft does not do any effort to make winforms multiplatform happen. Anyway , MAUI is something to wait a bit more than we expected. Until than I still using gambas3 for desktop apps

http://gambas.sourceforge.net/en/main.html

it is close to winforms for create desktop apps in linux

**Comment by weltkante** [comment]
The problem with adopting the mono implementation of WinForms is that its only superficial, a lot of low level details won't work or work differently, like window messages, but many complex controls require the low level details to be correct. If Microsoft were to "copy paste" the mono version of WinForms they would be expected to support and contribute to it to make things work. That would not be easy since the underlying platform WinForms is wrapping is not open source.

TLDR: picking up the mono implementation of WinForms is not a valid option because its not supportable by Microsoft in a straightforward way, it has to remain third party or Microsoft has to invest significant effort. If they have to invest significant effort anyways their choice was to build on Xamarin and not on WinForms.

**Comment by gsmental** [comment]
@OliaG BUT  linux support in maui?
